cfgov/unprocessed/js/molecules/Autocomplete.js

```diff
@@ -121,23 +130,60 @@ function Autocomplete( element, opts ) {
       inputCoords.left ) + 'px';
   }
 
+  /**
+   * Toggles the error state and message that's shown when the
+   * max search term length is hit
+   */
+  function _toggleMaxLengthError() {
+    _maxLengthExceeded = _searchTerm.length >= _settings.maxChars ? true :
+      false;
+    if ( _maxLengthExceeded ) {
+      _input.classList.add( ERROR_CLASS );
+
+      /* If you type fast enough, search results from a letter or two back can
+         cover up the max length error, so we'll wait a bit before hiding */
+      setTimeout( _hide, _settings.delay );
+    } else {
+      _input.classList.remove( ERROR_CLASS );
+    }
+    _instance.dispatchEvent( 'maxCharacterChange', {
+      maxLengthExceeded: _maxLengthExceeded,
+      searchTerm: _searchTerm
+    } );
+  }
+
+  /**
+   * Event handler for input into the _input element
+   * @param {InputEvent} event The input event object
+   */
+  function _handleInput( event ) {
+    _searchTerm = event.target.value;
+    if ( _searchTerm.length >= _settings.minChars &&
+         _searchTerm.length < _settings.maxChars ) {
+      if ( _maxLengthExceeded === true ) {
```

Comment:
Could this be moved to the top of the function like this? 

```
/**
 * Event handler for input into the _input element.
 * @param {InputEvent} event - The input event object.
 * @returns {boolean} False if maximum character limit was reached, true otherwise.
 */
function _handleInput( event ) {
  if ( _maxLengthExceeded === true ) {
    _toggleMaxLengthError();
    return false;
  }

  if ( _searchTerm.length >= _settings.minChars &&
       _searchTerm.length < _settings.maxChars ) {
    if ( _settings.url ) {
      _throttleFetch();
    } else {
      _settings.filterList( _settings.list );
    }
  } else if ( _searchTerm.length >= _settings.maxChars ) {
    _toggleMaxLengthError();
  } else {
    _hide();
  }
    
  return true;
}
```

Reply:
Fixed in 608db2c59260146ac4f6e5630f84471ed0d60758, although a little differently than what's suggested here. This bit…

```javascript
if ( _maxLengthExceeded === true ) {
    _toggleMaxLengthError();
}
```

…is used to handle the case where you've entered 50 characters, gotten the error, but then deleted one or more characters, so we need to remove the error state and message and start giving you autocomplete results again. Moving that `if` up to the top of `_handleInput` without `return false` does just that in a much simpler way than what I had before.